aptos-move/aptos-release-builder/src/components/framework.rs

```diff
@@ -35,23 +46,50 @@ pub fn generate_upgrade_proposals(is_testnet: bool) -> Result<Vec<(String, Strin
             .unwrap()
             .to_string();
 
+        let mut args = vec![
+            "run",
+            "--bin",
+            "aptos",
+            "--",
+            "governance",
+            "generate-upgrade-proposal",
+            "--account",
+            publish_addr,
+            "--output",
+            move_script_path.to_str().unwrap(),
+            "--package-dir",
+            package_path.to_str().unwrap(),
+        ];
+
+        if is_testnet {
+            args.push("--testnet");
+        }
+
+        // If this file is the first framework file being generated (if `result.is_empty()` is true),
+        // its `next_execution_hash` should be the `next_execution_hash` value being passed in.
+        // If the `result` vector is not empty, the current file's `next_execution_hash` should be the
+        // hash of the latest framework file being generated (the hash of result.last()).
+        // For example, let's say we are going to generate these files:
+        // 0-move-stdlib.move	2-aptos-framework.move	4-gas-schedule.move	6-features.move
+        // 1-aptos-stdlib.move	3-aptos-token.move	5-version.move		7-consensus-config.move
+        // The first framework file being generated is 3-aptos-token.move. It's using the next_execution_hash being passed in (so in this case, the hash of 4-gas-schedule.move being passed in mod.rs).
+        // The second framework file being generated would be 2-aptos-framework.move, and it's using the hash of 3-aptos-token.move (which would be result.last()).
+        let mut _execution_hash: String = "".to_owned();
+        if !next_execution_hash.is_empty() {
+            args.push("--next-execution-hash");
+            if result.is_empty() {
+                _execution_hash = next_execution_hash.to_owned();
+                args.push(&_execution_hash);
+            } else {
+                _execution_hash =
+                    HashValue::sha3_256_of(result.last().unwrap().1.as_bytes()).to_string();
+                args.push(&_execution_hash);
+            }
+        }
+
         assert!(Command::new("cargo")
             .current_dir(root_path.as_path())
-            .args(&vec![
-                "run",
-                "--bin",
-                "aptos",
-                "--",
-                "governance",
-                "generate-upgrade-proposal",
-                if is_testnet { "--testnet" } else { "" },
-                "--account",
-                publish_addr,
-                "--package-dir",
-                package_path.to_str().unwrap(),
-                "--output",
-                move_script_path.to_str().unwrap(),
-            ])
```

Comment:
gaaah, how did this ever exist, we should have used the structs / interfaces directly lol

Reply:
I implemented that... Was mostly because I moved them from a previous test and didn't have the time to look into the structs calls. Would be definitely so much better if we did it that way cuz it will save us the time for recompiling the binaries.